src/measure_theory/l2_space.lean

```diff
@@ -0,0 +1,144 @@
+/-
+Copyright (c) 2021 Rémy Degenne. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Rémy Degenne
+-/
+import analysis.normed_space.inner_product
+import measure_theory.set_integral
+
+/-! # `L^2` space
+
+If `E` is an inner product space over `𝕜` (`ℝ` or `ℂ`), then `Lp E 2 μ` (defined in `lp_space.lean`)
+is also an inner product space, with inner product defined as `inner f g = ∫ a, ⟪f a, g a⟫ ∂μ`.
+
+### Main results
+
+* `mem_L1_inner` : for `f` and `g` in `Lp E 2 μ`, the pointwise inner product `λ x, ⟪f x, g x⟫`
+  belongs to `Lp 𝕜 1 μ`.
+* `integrable_inner` : for `f` and `g` in `Lp E 2 μ`, the pointwise inner product `λ x, ⟪f x, g x⟫`
+  is integrable.
+* `L2.inner_product_space` : `Lp E 2 μ` is an inner product space.
+
+### Notation
+
+* `α →₂[μ] E` : the type `Lp E 2 μ`.
+
+-/
+
+noncomputable theory
+open topological_space measure_theory measure_theory.Lp
+open_locale nnreal ennreal
+
+notation α ` →₂[`:25 μ `] ` E := measure_theory.Lp E 2 μ
```

Comment:
The notation is great. Could you introduce this notation alongside the one for `L^1` in the file `lp_space`, and localize both of them to the `measure_theory` namespace? Otherwise, LGTM, thanks!

Reply:
The notations are now both defined inside of the `measure_theory` namespace. Is that enough to localize them?

Reply:
To localize them, you need an incantation like
```
localized "notation `ω` := omega_limit" in omega_limit
```
Then the notation is only available when you `open omega_limit`.

Reply:
It looks like https://leanprover.github.io/theorem_proving_in_lean/interacting_with_lean.html#notation is wrong on that point. It states that "If you declare these notations in a namespace, the notation is only available when the namespace is open." But I just tested it and you are right, I need to localize it explicitly.

If I localize the notation for L1 to `measure_theory`, I break l1_space, bochner_integration etc. That's because the `L1` namespace which is used in those files is not inside `measure_theory`.

Reply:
Found the trick. I needed to `open_locale measure_theory` in those files. `open measure_theory` is not enough.

Reply:
Yes, sorry, you need `open_locale` and not `open`.